src/time/Waveform.cpp

```diff
@@ -23,6 +23,7 @@ int Waveform::getDegree() const
 
 time::Storage &Waveform::timeStepsStorage()
 {
+  _recomputeBuffer = true;
```

Comment:
```suggestion
  _bspline.requireBufferUpdate();
```
Instead of maintaining `_recomputeBuffer` here, you can just tell `Bspline` that a buffer update will be required. I think this would look the following:
```
void Bspline::requireBufferUpdate() {
    _recomputeBuffer = true;
}
```